h2o-automl/src/main/java/ai/h2o/automl/ModelingStep.java

```diff
@@ -370,7 +370,8 @@ protected void setStoppingCriteria(Model.Parameters parms, Model.Parameters defa
         if (parms._stopping_metric == StoppingMetric.AUTO) {
             String sort_metric = getSortMetric();
             parms._stopping_metric = sort_metric == null ? StoppingMetric.AUTO
-                    : sort_metric.equals("auc") ? StoppingMetric.logloss
+                    : sort_metric.equals("auc")  ? StoppingMetric.logloss
+                    : sort_metric.equals("rmse") ? StoppingMetric.deviance
```

Comment:
> I believe similar cases could occur in other "distributions" so I think the safest choice for default regression stopping metric is deviance.

I think we're making AutoML even less predictable with this kind of change. And it looks like your justification in the description is mainly based on a corner case, so I'm pretty skeptical about this.

Reply:
For Gaussian the deviance is MSE so that should be "compatible" with the RMSE. And if the user specifies a distribution, I'd guess that they'd expect to use the metric that is typical for that distribution (i.e. deviance) not RMSE. To me stopping metric RMSE means that I want to stop when maximizing the Gaussian error distribution which is to me more confusing when I'd specified, e.g., tweedie.

RMSE is a sorting metric because deviance has different definitions for different distributions which would mean that the models in the leaderboard would end up being sorted by some metric with multiple definitions.

Does it makes more sense to you now?

Reply:
sorry, I overlooked the code, this is applied only when `stopping_metric` is set to AUTO, so it's not as confusing as I thought first.

I just remember that we wanted to simplify this logic because we don't even have any proper justification for the `auc -> logloss` case, so a bit surprise to see that we're extending this part now.

Reply:
No problem, the code is confusing (maybe I should use if-elses instead of the ternary operator there).

I think the reason for `logloss` was that it is "easier" to optimize it when compared to optimizing `auc` but I don't know if we have any evidence of that. But it also makes _some_ sense to me applying the stopping metric to a metric that is closer to what we optimize than to a sorting metric we use to compare models.

Reply:
Although, and to contradict what I said about the artificiality of this relation, it's also logical to optimize by default on the same metric that you want to use to compare models, because at the end of the day, you're mainly interested in the best model.
So by setting `sort_metric=auc` I'm sure most users expect that all models are optimized for that metric.

Reply:
> I think the ternary is clear this way ;)
(that's how I use it, indented a la pattern matching in Haskell: | cond -> action with : looking a bit like | and ? like a weird arrow)

Since I know lisp better than Haskell, I usually double-check things where operator precedence can make difference 😅. (At least in those cases less frequent cases.) But now I will remember `| cond -> action` 😄 .


> My point is that to clarify this logic I'm wondering it it wouldn't make more sense to completely remove this relation, and set stopping_metric if AUTO only based on task type | distribution instead of defaulting to sort_metric in some cases but not in others.

Makes sense, I will make the modifications.